### PR TITLE
fix comparing versions

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

here is some late update between importing pkg and getting pkg version, so Colab kernel needs to be restarted which is not very intuitive, this uses always the `package.__version__`

See in your Colab:
```
!pip install torchtext==0.6.0

print(torchtext.__version__)
# 0.6.0
from pkg_resources import get_distribution
print(get_distribution('torchtext').version)
# 0.9.0
```

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Comment by Borda** [comment]
@carmocca any idea why docs are failing even, I did not touch them... we say exactly the same error in the past but never figger out what causes it, right?

**Comment by carmocca** [comment]
> @carmocca any idea why docs are failing even, I did not touch them... we say exactly the same error in the past but never figger out what causes it, right?

In the previous PRs that produced this error, running `make docs` from the root directory worked and you mentioned that the difference was `SPHINX_MOCK_REQUIREMENTS`. Have you checked if `make docs` works for this PR?

File: pl_examples/__init__.py [hunk 1]
```diff
@@ -15,10 +15,10 @@
 _DATASETS_PATH = os.path.join(_PACKAGE_ROOT, 'Datasets')
 
 _TORCHVISION_AVAILABLE = _module_available("torchvision")
-_TORCHVISION_MNIST_AVAILABLE = True
+_TORCHVISION_MNIST_AVAILABLE = _TORCHVISION_AVAILABLE
 _DALI_AVAILABLE = _module_available("nvidia.dali")
 
-if _TORCHVISION_AVAILABLE:
+if _TORCHVISION_MNIST_AVAILABLE:
```
Comment (Borda) [inline pl_examples/__init__.py:21]:
```suggestion
if _TORCHVISION_AVAILABLE:
```

File: requirements/extra.txt [hunk 2]
```diff
@@ -7,4 +7,5 @@ torchtext>=0.5
 # onnx>=1.7.0
 onnxruntime>=1.3.0
 hydra-core>=1.0
+# todo: when user standard package stream, drop `fairscale` from hard mocked docs libs
```
Comment (Borda) [inline requirements/extra.txt:10]:
```suggestion
# todo: when switch to standard package stream, drop `fairscale` from hard mocked docs libs
```

**Pull request merged**